homework02/allan/string.cc

```diff
@@ -0,0 +1,103 @@
+#include "string.h"
```

Comment:
It seems that in your String implementation, `capacity` is always defined as `size + 1`. This will lead to surprises, as the capacity of string / STL container is the not the " total memory being used " but the actual number of _elements_ the string / container has allocated memory for. This means that if capacity() returns 10, I should be able to insert characters in the string without extra memory allocations up to 10 elements, which isn't true here.